p2p/transport/webrtc/deadline.go

```diff
@@ -0,0 +1,77 @@
+package libp2pwebrtc
+
+import (
+	"sync"
+	"time"
+)
+
+type deadline struct {
```

Comment:
This seems overly complicated, and wasteful. We should only start a timer when there’s actually a `Read` / `Write` call running (and stop the timer as soon as that call returns), not just because the deadline was set.

It also seems like we’re not using the deadline correctly (see comments in `datachannel.go`).

Reply:
This allows updating the deadline while the Read/Write call is running.

Reply:
Code has completely changed since, so if something similar is still an issue it will have to be rechecked in a new review I am afraid.

Reply:
As such, this comment can be resolved